clients/lsp-csharp.el

```diff
@@ -235,6 +241,99 @@ tarball or a zip file (based on a current platform) to TARGET-DIR."
   "Resolves path and arguments to use to start the server."
   (list (lsp-csharp--language-server-path) "-lsp"))
 
+(defun lsp-csharp--reset-test-buffer (present-buffer)
+  "Create new or reuse an existing test result output buffer.
+
+PRESENT-BUFFER will make the buffer be presented to the user."
+  (let ((existing-buffer (get-buffer lsp-csharp-test-run-buffer-name))
+        (project-root-dir (lsp--suggest-project-root)))
+    (if existing-buffer
+        (progn
+          (with-current-buffer existing-buffer
+            (setq buffer-read-only nil)
+            (erase-buffer)
+            (setq buffer-read-only t)
+            (setq default-directory project-root-dir))
+          existing-buffer)
+      (let ((buffer (get-buffer-create lsp-csharp-test-run-buffer-name)))
+        (with-current-buffer buffer
+          (setq default-directory project-root-dir)
```

Comment:
why this is needed?

Reply:
When in compliation mode file paths in error messages and such can be clicked-on/activated via keyboard -- as the paths are not normalized at FS root (`/`) -- this helps it to resolve the actual file. At least this is the behaviour in doom-emacs configuration I have -- not sure if that is the standard behaviour.

I.e. test run buffer contains output from the compiler as well (warnings+errors), not just the actual output from tests.

Reply:
There might be a better way, right.

Reply:
```emacs-lisp
(with-current-buffer (get-buffer-create lsp-csharp-test-buffer-name)
  (compilation-mode)
  (read-only-mode 1)
  (setq-local default-directory (lsp-workspace-root))
  (let ((inhibit-read-only t))
    (erase-buffer)
    (insert message "\n"))
  (current-buffer))
```

Reply:
Another thing that could be done is to make a special major-mode for this, deriving from compilation-mode so that users can set-up custom keybindings. I.e.
```emacs-lisp
(define-derived-mode lsp-csharp-test-mode compilation-mode
  "Major mode for running `lsp-csharp` tests"
  (read-only-mode 1)
  (setq-local default-directory (lsp-workspace-root))
```
And then the first three lines (compilation-mode ...) could simply be `(lsp-csharp-test-mode)`.